pallets/rmrk-core/src/functions.rs

```diff
@@ -291,20 +304,76 @@ where
 					royalty = Some(RoyaltyInfo::<T::AccountId> { recipient, amount });
 				},
 				None => {
-					royalty =
-						Some(RoyaltyInfo::<T::AccountId> { recipient: owner.clone(), amount });
+					// If a royalty amount is passed but no recipient, defaults to the sender
+					royalty = Some(RoyaltyInfo::<T::AccountId> { recipient: sender, amount });
 				},
 			}
 		};
 
-		let owner_as_maybe_account = AccountIdOrCollectionNftTuple::AccountId(owner.clone());
+		let nft = NftInfo {
+			owner: AccountIdOrCollectionNftTuple::AccountId(owner),
+			royalty,
+			metadata,
+			equipped: false,
+			pending,
+			transferable,
+		};
+
+		Nfts::<T>::insert(collection_id, nft_id, nft);
+
+		// increment nfts counter
+		let nfts_count = collection.nfts_count.checked_add(1).ok_or(ArithmeticError::Overflow)?;
+		Collections::<T>::try_mutate(collection_id, |collection| -> DispatchResult {
+			let collection = collection.as_mut().ok_or(Error::<T>::CollectionUnknown)?;
+			collection.nfts_count = nfts_count;
+			Ok(())
+		})?;
+
+		Ok((collection_id, nft_id))
+	}
+
+	fn nft_mint_directly_to_nft(
+		sender: T::AccountId,
+		owner: (CollectionId, NftId),
+		collection_id: CollectionId,
+		royalty_recipient: Option<T::AccountId>,
+		royalty_amount: Option<Permill>,
+		metadata: StringLimitOf<T>,
+		transferable: bool,
+	) -> sp_std::result::Result<(CollectionId, NftId), DispatchError> {
+		let nft_id = Self::get_next_nft_id(collection_id)?;
+		let collection = Self::collections(collection_id).ok_or(Error::<T>::CollectionUnknown)?;
+
+		// Prevent minting when next NFT id is greater than the collection max.
+		if let Some(max) = collection.max {
+			ensure!(nft_id < max, Error::<T>::CollectionFullOrLocked);
+		}
+
+		// Calculate the rootowner of the intended owner of the minted NFT
+		let rootowner = Self::lookup_root_owner(owner.0, owner.1)?.0;
```

Comment:
More of nitpick since other places in code use this syntax.
```
let (rootowner, _) = Self::lookup_root_owner(owner.0, owner.1)?;
```

Reply:
please do nitpick, fixed in [282101f](https://github.com/rmrk-team/rmrk-substrate/pull/178/commits/282101f2a96d138e0133c65ade1d64515029ac57)